src/world/shipparts/heal.rs

```diff
@@ -0,0 +1,60 @@
+use crate::timer::Timer;
+use crate::world::types::{Location, Module, ModuleInputs, WorldEvent};
+use mlua::prelude::{LuaFunction, LuaTable};
+use mlua::{Lua, RegistryKey, Result, ToLua, UserData, UserDataFields, UserDataMethods};
+
+struct Heal {
+    timer: Timer,
+    hull: RegistryKey,
+}
```

Comment:
heal.rs is quite a bit bigger than heal.rs, but most of the bulk and complexity comes from being between two layers of Lua. It needs to call hull.repair, which is a Lua function.